src/schema/schema.js

```diff
@@ -196,23 +198,30 @@ function getMutationField(graffitiModel, type, viewer) {
 function getFields(graffitiModels, {mutation = true} = {}) {
   const types = getTypes(graffitiModels);
 
+  const GraphQLViewer = new GraphQLObjectType({
+    name: 'Viewer',
+    fields: reduce(types, (fields, type, key) => {
+      type.name = type.name || key;
+      const graffitiModel = graffitiModels[type.name];
+      return {
+        ...fields,
+        ...getConnectionField(graffitiModel, type),
+        ...getSingularQueryField(graffitiModel, type)
+      };
+    }, {
+      id: globalIdField('Viewer')
+    }),
+    interfaces: [nodeInterface]
+  });
+  // register viewer type
+  addType('Viewer', GraphQLViewer);
+
   const viewer = {
     name: 'viewer',
-    type: new GraphQLObjectType({
-      name: 'Viewer',
-      fields: reduce(types, (fields, type, key) => {
-        type.name = type.name || key;
-        const graffitiModel = graffitiModels[type.name];
-        return {
-          ...fields,
-          ...getConnectionField(graffitiModel, type),
-          ...getSingularQueryField(graffitiModel, type)
-        };
-      }, {
-        id: idField
-      })
-    }),
-    resolve: () => ({id: 'viewer'})
+    type: GraphQLViewer,
+    resolve: () => {
+      return new Viewer()
```

Comment:
This will return a new Viewer on every request, do you think this is the correct behaviour?